### ISSUE TITLE
no-typos: destructured required propType throws an error

### ISSUE THREAD

**Issue by AlexMarvelo** [yannickcr/eslint-plugin-react#1489]
`no-typos` is a very nice rule, it helps a lot. But I have one case, where it throws useless error (and I have hundreds of them around my project)

1. This is pattern is considered a warning:
```
const { bool } = PropTypes;
class MyComponent extends React.Component {
  static propTypes = {
    a: bool.isRequired
  }
}
```

2. And this is not:
```
class MyComponent extends React.Component {
  static propTypes = {
    a: PropTypes.bool.isRequired
  }
}
```

Do you agree, that case 1 shouldn't throw any errors?

**Comment by ljharb** [comment]
Duplicate of #1389

**Comment by alex-shamshurin** [comment]
Yes, I have the same issue. It works inside React.component extended class and doesn't work for stateless components
